File: resthub-apps/booking-js/src/main/java/org/resthub/booking/dao/HotelDao.java
package org.resthub.booking.dao;

import org.resthub.booking.model.Hotel;
import org.resthub.core.dao.GenericResourceDao;
import org.synyx.hades.domain.Page;
import org.synyx.hades.domain.Pageable;


/**
 * Poll DAO.
 * @author Nicolas Carlier
 */
public interface HotelDao extends GenericResourceDao<Hotel>  {

	/**
     * Find hotel by fulltext search.
     * @param query query
	 * @param pageable
     * @return hotels matching query or null if query string is in an incorrect format.
     */
    Page<Hotel> find(String query, Pageable pageable);

    /**
     * Rebuil full index.
     */
    void rebuildIndex();
}

File: resthub-apps/booking/booking-js/src/main/webapp/models/hotel.dao.js
define([ 'jquery.dao' ], function(CrudDao) {
	return CrudDao.define("HotelDao")
		.supplement({
		root : 'api/hotel/',

		find : function(callback, val, page, size) {
			var url = this.root + 'search?page=' + page + '&size=' + size;
			if (val) {
				url = url + '?q=' + val;
			}
			this._get(url, callback);
		}
	});
});
